### PR TITLE
Fix #966

### LINKED ISSUE #966: control.Postgres does not try to use subcommand if psychopg fails.
A regression test must be introduced, as this is not getting caught currently.

aiida/contol/postgres.py:103:

```
if not self.pg_execute:
```

should be 

```
if self.pg_execute == _pg_execute_not_connected
```

### PR DESCRIPTION
**Pull request by DropD**
fix #966 

* Quicksetup now again falls back on psql subcommands if psychopg can not be used to connect.
* The error was in `control.postgres.Postgres.determine_setup` (Thanks to @ltalirz for finding it!)
* Regression test added in `control.tests.test_postgres`.

### CODE REVIEW

**Comment by ltalirz** [comment]
By the way: this also needs to go into 0.10.1

**Comment by DropD** [comment]
How do we get this into the release branch? Will develop get merged into there or should i rebase and make a pr?

**Comment by ltalirz** [comment]
I think you need to start again from 0.10.1
I've been told develop will go into 0.11, but I guess it won't go  into 0.10.0

**Comment by sphuber** [comment]
Yes, if you want this to end up in `0.10.1` as well, you should have rebased it onto the release branch, but since you already merged it into develop, you have to make a new PR. The commit should normally end up in develop once the release is made, after which it will be merged into develop. Now that commit will already be there, but I doubt that will lead to issues.  

TLDR: if you want it in `0.10.1`, make a new PR based of `release_v0.10.1`

**Comment by ltalirz** [comment]
You can certainly not simply remove the second if clause, since the first if clause may change `self.pg_execute`.
Anyhow, in my opinion this code is a bit hard to read.

**Comment by szoupanos** [comment]
I agree with both comments
